testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -1345,6 +1345,169 @@ void testCastStringToDateTime(CastType castType, SqlOperatorFixture f) {
+      f.checkString("cast(time '12:30:55' as varchar format 'PM')",
+          "PM",
+          "VARCHAR NOT NULL");
+    }
+
+    // Cast TIMESTAMP to String
+    if (Bug.CALCITE_6269_FIXED) {
+      // Query output cannot be validated as it's dependent on execution time zone
+      f.checkQuery("cast(timestamp '2008-12-25 00:00:00+06:00' as varchar format 'TZH')");
+      f.checkString("cast(timestamp '2008-12-25 00:00:00+00:00' as varchar format "
+              + "'TZM' AT TIME ZONE 'Asia/Kolkata')",
+          "30",
+          "VARCHAR NOT NULL");
+    }
+  }
+
+  @Test void testCastFormatClauseStringToDateTime() {
+    final SqlOperatorFixture f = fixture()
+        .withLibrary(SqlLibrary.BIG_QUERY)
+        .setFor(SqlStdOperatorTable.CAST);
+    f.checkScalar("cast('18-12-03' as date format 'YY-MM-DD')",
+        "2018-12-03",
+        "DATE NOT NULL");
+    f.checkScalar("cast('JUN 30, 2018' as date format 'MON DD, YYYY')",
+        "2018-06-30",
+        "DATE NOT NULL");
+    f.checkScalar("cast('17:30' as time format 'HH12:MI')",
+        "17:30:00",
+        "TIME(0) NOT NULL");
+    f.checkScalar("cast('01:05:07.16' as time format 'HH24:MI:SS.FF4')",
+        "01:05:07",
+        "TIME(0) NOT NULL");
+    f.checkScalar("cast('2017-05-12' as timestamp format 'YYYY-MM-DD')",
+        "2017-05-12 00:00:00",
+        "TIMESTAMP(0) NOT NULL");
+    f.checkScalar("cast('2020.06.03 12:42:53' as timestamp format 'YYYY.MM.DD HH:MI:SS')",
+        "2020-06-03 12:42:53",
+        "TIMESTAMP(0) NOT NULL");
+
+    if (Bug.CALCITE_6269_FIXED) {
+      f.checkScalar("cast('2020.06.03 00:00:53+06:30' as timestamp format"
+              + " 'YYYY.MM.DD HH24:MI:SSTZH:TZM')",
+          "2020-06-02 17:30:53 UTC",
+          "TIMESTAMP(0) NOT NULL");
+      f.checkScalar("cast('03:30 P.M.' as time format 'HH:MI P.M.')",
+          "15:30:00",
+          "TIME(0) NOT NULL");
+    }
+  }
+
+  @Test void testCastFormatClauseByteToString() {
+    final SqlOperatorFixture f = fixture()
+        .withLibrary(SqlLibrary.BIG_QUERY)
+        .setFor(SqlStdOperatorTable.CAST);
+
+    if (Bug.CALCITE_6270_FIXED) {
+      f.checkString("cast(b'\\x48\\x65\\x6c\\x6c\\x6f' as varchar format 'ASCII')",
+          "Hello",
+          "VARCHAR");
+      f.checkScalar("cast('Hello' as varbinary format 'ASCII')",
+          "\\x48\\x65\\x6c\\x6c\\x6f",
+          "VARBINARY NOT NULL");
+    }
+  }
+
+  @Test void testCastFormatClauseNumericToString() {
+    final SqlOperatorFixture f = fixture()
+        .withLibrary(SqlLibrary.BIG_QUERY)
+        .setFor(SqlStdOperatorTable.CAST);
+
+    if (Bug.CALCITE_6270_FIXED) {
+      f.checkString("cast(-12.23 as varchar FORMAT '999.999')",
+          "-12.230",
+          "VARCHAR NOT NULL");
+      f.checkString("cast(1234.56 as varchar FORMAT '$999,999.999')",
+          "$1,234.560",
+          "VARCHAR NOT NULL");
+      f.checkString("cast(123456 as varchar FORMAT '9.999EEEE')",
+          "1.235E+05",
```

Comment:
Does adding a test for `SAFE_CAST` give us any value here?

Reply:
Essentially both operators should take the same flow, I will add some tests to see it there's any deviation in functionality.

Reply:
Ok no need to make this PR any bigger if it doesn't give us anything so it's up to you

Reply:
Changed all the tests to use `safeParameters()` to test all CAST functions.